config/src/main/java/com/typesafe/config/ConfigResolveOptions.java

```diff
@@ -91,7 +96,24 @@ public boolean getUseSystemEnvironment() {
      * @since 1.2.0
      */
     public ConfigResolveOptions setAllowUnresolved(boolean value) {
-        return new ConfigResolveOptions(useSystemEnvironment, value);
+        return new ConfigResolveOptions(useSystemEnvironment, value, resolver);
+    }
+
+    /**
+     * Returns options where the given resolver used as a fallback if a
+     * reference cannot be otherwise resolved.
+     */
+    public ConfigResolveOptions withFallbackResolver(ConfigReferenceResolver fallback) {
+        return new ConfigResolveOptions(useSystemEnvironment, allowUnresolved,
+                this.resolver.withFallback(fallback));
+    }
+
+    /**
+     * Returns the resolver to use as a fallback if a reference cannot be
+     * otherwise resolved. Never returns null.
+     */
+    public ConfigReferenceResolver getFallbackResolver() {
+        return this.resolver;
```

Comment:
The method names here could follow the conventions for `ConfigIncluder` in parse options perhaps; would be `appendResolver` and `getResolver` I believe.